File: Ruby/shopping_cart.rb
class ShoppingCart
  def initialize
    #
    # the goal is to remove this field, replacing with
    # @prices = []
    #

    @price = 0
  end

  def add(price)
    @price = price
  end

  def number_of_products
    1
  end

  def calculate_total_price
    @price
  end

  def has_discount?
    @price >= 100
  end
end

class SomeConsumer
  def do_stuff
    shopping_cart = ShoppingCart.new
    shopping_cart.add(100)
    puts "other consumer: #{shopping_cart.calculate_total_price}"
  end
end

if __FILE__ == $PROGRAM_NAME
  shopping_cart = ShoppingCart.new
  shopping_cart.add(10)
  puts "number of products: #{shopping_cart.number_of_products}"
  puts "total price: #{shopping_cart.calculate_total_price}"
  puts "has discount: #{shopping_cart.has_discount?}"
end

File: PHP/src/ShoppingCart.php
<?php
declare(strict_types=1);

namespace ParallelChange;

class ShoppingCart {

	// The goal is to remove this field, replacing it with
	// private $prices = [];
	private $price = 0;

	public function add(int $price) {
		$this->price = $price;
	}

	public function getNumberOfProducts(): int {
		return 1;
	}

	public function calculateTotalPrice(): int {
		return $this->price;
	}

	public function hasDiscount(): bool {
		return $this->price >= 100;
	}
}